### ISSUE TITLE
print order does not know about pseries

### ISSUE THREAD

**Issue by rwst** [pynac/pynac#22]
See http://trac.sagemath.org/ticket/17321
Needed by http://trac.sagemath.org/ticket/17659
```
sage: ex = sin(x).series(x==0, 3) * x
sage: str(ex)

Breakpoint 4, GiNaC::print_order::compare (this=0x7fffffffae90, lh=..., rh=...)
    at order.cpp:280
280                     else throw (std::runtime_error("comparing typeid's"));

#0  GiNaC::print_order::compare (this=0x7fffffffae90, lh=..., rh=...) at order.cpp:280
#1  0x00007fffcc8bf85b in GiNaC::print_order::compare (this=0x7fffffffae90, lhex=..., 
    rhex=...) at order.cpp:119
#2  0x00007fffcc8bf8a7 in GiNaC::print_order_pair_mul::operator() (
    this=0x7fffffffaf50, lhex=..., rhex=...) at order.cpp:124
#3  0x00007fffcc8585b1 in std::__push_heap<__gnu_cxx::__normal_iterator<GiNaC::expair*, std::vector<GiNaC::expair, std::allocator<GiNaC::expair> > >, long, GiNaC::expair, GiNaC::print_order_pair_mul> (__first=..., __holeIndex=1, __topIndex=0, __value=..., 
    __comp=...) at /usr/include/c++/4.8/bits/stl_heap.h:183
#4  0x00007fffcc857e0c in std::__adjust_heap<__gnu_cxx::__normal_iterator<GiNaC::expair*, std::vector<GiNaC::expair, std::allocator<GiNaC::expair> > >, long, GiNaC::expair, GiNaC::print_order_pair_mul> (__first=..., __holeIndex=1, __len=2, __value=..., 
    __comp=...) at /usr/include/c++/4.8/bits/stl_heap.h:325
#5  0x00007fffcc857b03 in std::make_heap<__gnu_cxx::__normal_iterator<GiNaC::expair*, st
etc

(gdb) p typeid_lh
$3 = (const GiNaC::tinfo_t) 0x7fffcc8c548c <GiNaC::symbol::tinfo_static>
(gdb) p typeid_rh
$4 = (const GiNaC::tinfo_t) 0x7fffcc8c4f18 <GiNaC::pseries::tinfo_static>
```
See also #32.

**Comment by rwst** [comment]
In current Sage:
```
sage: ex = sin(x).series(x==0, 3) * x
sage: ex
x*(1*x + Order(x^3))
sage: str(ex)
'x*(1*x + Order(x^3))'
```

**Comment by rwst** [comment]
This was not really fixed with 50c72277d3 because now print order is according to hash value which gves erratic results in doctests.
